test/test_force.js

```diff
@@ -118,63 +134,93 @@ function main(args) {
     }
 
     // Test all the platforms / app types requested
-    for (var i=0; i<chosenOperatingSystems.length; i++) {
-        var os = chosenOperatingSystems[i];
-        if (testingWithAppType) {
-            for (var j=0; j<chosenAppTypes.length; j++) {
-                var appType = chosenAppTypes[j];
-                createCompileApp(tmpDir, os, appType, null, pluginRepoUri, useSfdxRequested);
+    if (testingWithClis) {
+        for (var i=0; i<chosenClis.length; i++) {
+            var cliName = chosenClis[i];
+            var templates = templateHelper.getTemplates(SDK.forceclis[cliName]);
+            for (var j=0; j<templates.length; j++) {
+                var template = templates[j];
+                for (var k=0; k<template.platforms.length; k++) {
+                    var os = template.platforms[k];
+                    createCompileApp(tmpDir, os, template.appType, template.url, null, useSfdxRequested);
```

Comment:
Generating an app for all the platforms supported by the templates available for the chosen CLIs.